### PR TITLE
[SAPRK-20785][WEB-UI][SQL]Spark should provide jump links and add (count) in the SQL web ui.

### PR DESCRIPTION
**Pull request by guoxiaolongzte**
## propose:
it provide links that jump to Running Queries,Completed Queries and Failed Queries.
it add (count) about Running Queries,Completed Queries and Failed Queries.
This is a small optimization in in the SQL web ui.

## fix before:
before

## fix after:
after

## How was this patch tested?

manual tests

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by guoxiaolongzte** [comment]
@srowen @ajbozarth
Help to review the code, thanks.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: given this val includes `content` I'm not sure `summary` is the best name here.

**Review by ajbozarth — COMMENTED** [review]
Sounds like a good idea, just a small nit on the code, but UI changes like this need screenshots in the description so we can see the changes. Screenshots pending this looks good though.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: Rebuild, optimize the variable name. 
I add two screenshots.

**Comment by guoxiaolongzte** [comment]
@ajbozarth 
Rebuild, optimize the variable name.
I add two screenshots.Thanks.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: I'd rather not switch to a `var` (it's very un-scala), see below for alt suggestion

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: perhaps leave this as `summary`, but not `++ content` at the end

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: then you could replace `content` here with `summary ++ content`

**Review by ajbozarth — COMMENTED** [review]
Thanks for the updates (screenshots look good) some followup comments:

**Comment by guoxiaolongzte** [comment]
@ajbozarth 
Thank you very much for the suggestion that I have modified.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: have you run the style checker? I think this may be in the wrong order

Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: How to run the style checker? But i build the code with maven success.

Comment (ajbozarth) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: I can't remember what flags/options run the style check with mvn, but you can always run it directly with `dev/scalastyle`

Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: please see
scala

**Review by ajbozarth — APPROVED** [review]
LGTM

**Comment by guoxiaolongzte** [comment]
Thank you, I will work better for Spark web ui.
jenkins to test.

**Comment by guoxiaolongzte** [comment]
@srowen 
Help to merge to master and modify the status about https://issues.apache.org/jira/browse/SPARK-20785.Thank you.

**Comment by guoxiaolongzte** [comment]
@HyukjinKwon 
Help to merge to master and modify the status about https://issues.apache.org/jira/browse/SPARK-20785.Thank you.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: I would do this with string interpolation but I guess it is not a big deal.

**Comment by HyukjinKwon** [comment]
@guoxiaolongzte I can't. I am just a contributor like yo. I could say +1 :).

**Comment by guoxiaolongzte** [comment]
@HyukjinKwon  Thanks.
@srowen  Help to merge to master and modify the status about https://issues.apache.org/jira/browse/SPARK-20785.Thanks.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: @guoxiaolongzte What about addressing this comment too if you don't disagree with this?

Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: I'm sorry, I forgot to see this, I immediately modified, I think you said very reasonable.

Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: "Running Queries (${listener.getRunningExecutions.size})"  replace "Running Queries (" + listener.getRunningExecutions.size + ")"?

Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: Yes, I thought `s"Running Queries (${listener.getRunningExecutions.size})"` one is slightly better.

Comment (guoxiaolongzte) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: Thanks.I will have to modify and test it.

**Comment by guoxiaolongzte** [comment]
@SparkQA 
please test it.

**Comment by guoxiaolongzte** [comment]
@srowen Help to merge to master and modify the status about https://issues.apache.org/jira/browse/SPARK-20785.Thanks.

**Comment by guoxiaolongzte** [comment]
@srowen 
Help to merge to master and modify the status about https://issues.apache.org/jira/browse/SPARK-20785.Thanks.

**Comment by srowen** [comment]
I don't think those links are worth the vertical space they chew up

**Comment by guoxiaolongzte** [comment]
@srowen 
I think the style exists to provide links that jump to Running Queries or Completed Queries because it may be well below the fold if there are many workers.Job page, stage page, application page, master page, worker page and other pages, but also this style.

**Comment by srowen** [comment]
Yes I understand that, but I don't think the reasoning is as compelling here, and the style is a bit different.

**Comment by guoxiaolongzte** [comment]
@srowen 
SQL ui is really different from other ui. In the follow-up version, I will strive to improve the function of SQL ui.

**Comment by guoxiaolongzte** [comment]
@srowen 
Sir, what about the PR need to be modified?

**Comment by guoxiaolongzte** [comment]
@srowen 
Help to merge to master , Thanks.

**Comment by srowen** [comment]
I don't support this but don't object of someone else merges

**Comment by guoxiaolongzte** [comment]
@srowen
Sir, if so, I do not know how to deal with it.
But I really feel like this change, which makes spark SQL UI good. If you agree with this JIRA, I also plan to add this function to this SQL ui. I believe that spark SQL ui will be more powerful and easy to use for users. Thanks.

**Comment by guoxiaolongzte** [comment]
@gatorsmile
Help to merge to master , Thanks.

**Comment by guoxiaolongzte** [comment]
@gatorsmile
Help to merge to master , Thanks.

**Comment by HyukjinKwon** [comment]
Hi @jerryshao, do you maybe have a preference? To me, I don't have a strong preference but I think I am okay with this.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (jerryshao) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: Is the indention here correct? This seems a little weird to me.

**Comment by jerryshao** [comment]
Jenkins, retest this please.

**Comment by jerryshao** [comment]
Yes, I'm fine with it. @ajbozarth would you please take another look on this PR? Thanks.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala
Comment (gatorsmile) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/ui/AllExecutionsPage.scala]: Please follow the style in the other files of the package `org.apache.spark.sql.execution.ui`

**Comment by guoxiaolongzte** [comment]
This Jenkins error is not caused by this PR.

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by guoxiaolongzte** [comment]
And failed, who submitted the code in question.

**Comment by ajbozarth** [comment]
Still LGTM

**Comment by guoxiaolongzte** [comment]
@HyukjinKwon
Help merge to master.

**Comment by HyukjinKwon** [comment]
I think @jerryshao is an active committer who knows this one better than me. Since he is here, let me leave it to him.

**Comment by jerryshao** [comment]
There's still left comment not addressed.

**Comment by guoxiaolongzte** [comment]
Sorry, I accidentally deleted the code branch. I'm going to close this PR. I created a new PR https://github.com/apache/spark/pull/19346, which was modified and created based on the latest code.
In the new PR I fix the indentation problem.
Thanks.

**Pull request closed without merging**
